utils.c

```diff
@@ -305,3 +307,59 @@ char *janus_address_to_ip(struct sockaddr *address) {
 	}
 	return addr ? g_strdup(addr) : NULL;
 }
+
+/* PID file management */
+static char *pidfile = NULL;
+static int pidfd = -1;
+static FILE *pidf = NULL;
+int janus_pidfile_create(const char *file) {
+	if(file == NULL)
+		return 0;
+	pidfile = g_strdup(file);
+	/* Try creating a PID file (or opening an existing one) */
+	pidfd = open(pidfile, O_RDWR|O_CREAT, 0644);
+	if(pidfd < 0) {
+		JANUS_LOG(LOG_FATAL, "Error opening/creating PID file %s, does Janus have enough permissions?\n", pidfile);
+		return -1;
+	}
+	pidf = fdopen(pidfd, "r+");
+	if(pidf == NULL) {
+		JANUS_LOG(LOG_FATAL, "Error opening/creating PID file %s, does Janus have enough permissions?\n", pidfile);
+		close(pidfd);
+		return -1;
+	}
+	/* Try locking the PID file */
+	int pid = 0;
+	if(flock(pidfd, LOCK_EX|LOCK_NB) < 0) {
+		fscanf(pidf, "%d", &pid);
```

Comment:
Yep but we don't care, we just use it to fill in an informational pid integer (if it fails, e.g., empty file, it stays 0).